lib/ansible/modules/cloud/vmware/vsphere_copy.py

```diff
@@ -170,7 +170,7 @@ def main():
         try:
             if isinstance(e[0], int):
                 error_code = e[0]
-        except KeyError:
+        except:
```

Comment:
Rather than bare exception, can we add IndexError as well ? 

```
except (KeyError, IndexError) as e:
```

Reply:
No, the real solution is to know which type of Exception (in the previous try-catch block) returns an error as a list. We know socket.error is in this case, but it seems another type of exception does this as well.

So we should catch that Exception as the right type, and handle it accordingly (and not bother with handling the general Exception type). I would even remove that `except Exception as e` part completely. (And test different type of error conditions to see what other type of Exceptions are common).

Reply:
I don't see how swallowing an error can ever be correct, first the code should never swallow, then we can try and extract as much detail out of it as possible (but is that really necessary ?)

Reply:
@moshloop I don't understand why you think I am promoting to swallow errors. The problem here is that we want to handle all kinds of exceptions where we have no clue how each represents its data. This has lead to the above junk code.

The solution is to only catch the specific exception exactly how it represents data.

Reply:
@moshloop Did I not propose that ??

> I would even remove that except Exception as e part completely. (And test different type of error conditions to see what other type of Exceptions are common).

Reply:
i.e.
```
    except:
        module.fail_json(msg=to_native(e), status=None, reason=to_native(e), url=url, exception=traceback.format_exc())
```

I don't think testing different error conditions is really feasible (e.g network drops, datastore full, datastore disconnected, datastore connected but misbehaving,  permissions, etc..)

Reply:
@moshloop VMware does not have specific Exceptions for each of those, do they ? And if there are a few, you can list them together, or use the base class of which the more specific Exceptions are built from.

What we definitely don't want to have is a blank `except:` statement, because we know it leads to issues causing tracebacks.

Reply:
This bit of code is piggy-backing off `module_utils.urls.open_url` so I am not even sure why it is raising an exception at all  - either way we can't attempt to handle all conceivable HTTP error conditions, open_url seems to handle the most common.

Reply:
It didn't used to use open_url(), so likely the exception-handling should be reviewed. Again, I would advise to remove the whole block, and start adding very specific exceptions that have been triggered. At the same time we can add integration tests for each of these exceptions and test the error messages.